apps/smart-contracts/token/test/MiniSales.test.ts

```diff
@@ -435,4 +455,180 @@ describe('=> MiniSales', () => {
       await expect(tx).to.emit(miniSales, 'PurchaseHookChange').withArgs(JUNK_ADDRESS)
     })
   })
+
+  describe('# testEndToEnd', () => {
+    let disallowedSourceUser: SignerWithAddress
+    let disallowedDestinationUser: SignerWithAddress
+    let blockedRecipient: SignerWithAddress
+    let ppo: PPO
+    let allowlistPurchaseHook: AllowlistPurchaseHook
+    let sourceAllowlist: AccountList
+    let destinationAllowlist: AccountList
+    let blocklist: AccountList
+    let purchaseAllowlist: AccountList
+    let restrictedTransferHook: RestrictedTransferHook
+    let allowedSourceUser: SignerWithAddress
+    let allowedDestinationUser: SignerWithAddress
+    let blockedUser: SignerWithAddress
+    let unblockedUser1: SignerWithAddress
+    let allowedPurchaser: SignerWithAddress
+
+    const getSignersAndMockTokenForEndToEndStack = async (): Promise<void> => {
```

Comment:
```suggestion
    const getSignersAndMockTokenForEndToEndTest = async (): Promise<void> => {
```
Technically these users are not being setup for the stack, but for end to end testing.